### PR TITLE
[AMDGPU][MC] Instructions not to be supported in GFX940

### LINKED ISSUE #55591: [AMDGPU][MC][GFX940] buffer_store_lds_dword, buffer_wbinvl1 and buffer_wbinvl1_vol should not be supported
AFAIK, these opcodes are not supported on GFX940.
Examples of failed tests:

    buffer_wbinvl1
    buffer_wbinvl1_vol
    buffer_store_lds_dword s[4:7], s2 offset:4095 lds

@rampitec, should these be disabled?

### PR DESCRIPTION
**Pull request by jwanggit86**
Buffer_store_lds_dword, buffer_wbinvl1, and buffer_wbinvl1_vol are obsolete in GFX940 and should not be supported.

### CODE REVIEW

File: llvm/lib/Target/AMDGPU/BUFInstructions.td [hunk 1]
```diff
@@ -3277,12 +3277,18 @@ defm BUFFER_ATOMIC_XOR_X2       : MUBUF_Real_Atomic_vi <0x6a>;
 defm BUFFER_ATOMIC_INC_X2       : MUBUF_Real_Atomic_vi <0x6b>;
 defm BUFFER_ATOMIC_DEC_X2       : MUBUF_Real_Atomic_vi <0x6c>;
 
-defm BUFFER_STORE_LDS_DWORD     : MUBUF_Real_vi_gfx90a <0x3d>;
+let AssemblerPredicate = isGFX8GFX9NotGFX90A in {
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/BUFInstructions.td:3280]: You generally shouldn't need to touch AssemblerPredicate. The SubtargetPredicate on the pseudodefintion is wrong, should fix it there.

**Review by arsenm — COMMENTED** [review]
Are BUFFER_WBL2 and BUFFER_INVL2 also removed and mishandled?

**Comment by jwanggit86** [comment]
> Are BUFFER_WBL2 and BUFFER_INVL2 also removed and mishandled?

Based on the ISA doc, `buffer_wbl2` is supported in GFX940, while `buffer_invl2` is not. Sp3, however, allows `buffer_invl2` for GFX940.

**Comment by arsenm** [comment]
> Sp3, however, allows `buffer_invl2` for GFX940.

This sounds more like an sp3 bug

File: llvm/lib/Target/AMDGPU/BUFInstructions.td [hunk 2]
```diff
@@ -1132,7 +1132,7 @@ let OtherPredicates = [HasGFX10_BEncoding] in {
   >;
 }
 
-let SubtargetPredicate = isGFX8GFX9 in {
+let SubtargetPredicate = isGFX8GFX9NotGFX940 in {
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/BUFInstructions.td:1135]: These clumsy subtarget checks should really be replaced with a subtarget feature

**Review by arsenm — APPROVED** [review]
lgtm but I think we should add more subtarget features instead of these increasingly complex generation checks

**Pull request merged**